src/streaming-client.ts

```diff
@@ -88,7 +93,10 @@ export class RevAiStreamingClient extends EventEmitter {
         }
 
         this.client.connect(url);
-        return new BufferedDuplex(this.requests, this.responses, { readableObjectMode: true });
+        return new BufferedDuplex(this.requests, this.responses, {
+            readableObjectMode: true,
+            writableObjectMode: true
```

Comment:
This is part of the proposed fix, enabling `writableObjectMode`. If "EOS" is written to the stream, when reading the chunk, it would return a buffer of length 3. If "EOS" bytes were present in the binary data from media, the read chunk would be longer than 3.